source/slang/slang-stdlib.cpp

```diff
@@ -217,19 +217,19 @@ namespace Slang
         { kIROp_Mul,     "*",    ARITHMETIC_MASK },
         { kIROp_Div,     "/",    ARITHMETIC_MASK },
         { kIROp_Mod,     "%",    INT_MASK },
-        { kIROp_And,     "&&",   LOGICAL_MASK },
-        { kIROp_Or,      "||",   LOGICAL_MASK },
+        { kIROp_And,     "&&",   BOOL_MASK | BOOL_RESULT},
+        { kIROp_Or,      "||",   BOOL_MASK | BOOL_RESULT },
```

Comment:
Same here; if we use `BOOL_MASK` then `BOOL_RESULT` shouldn't matter, if I'm understanding things correctly.